pkg/autoscaling/metrics/prometheus.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/pingcap/log"
+	"github.com/pkg/errors"
+	promClient "github.com/prometheus/client_golang/api"
+)
+
+const (
+	tikvSumCPUUsageMetricsPattern = `sum(increase(tikv_thread_cpu_seconds_total{cluster="%s"}[%s])) by (instance)`
+	tidbSumCPUUsageMetricsPattern = `sum(increase(process_cpu_seconds_total{cluster="%s",job="tidb"}[%s])) by (instance)`
+	tikvSumCPUQuotaMetricsPattern = `tikv_server_cpu_cores_quota{cluster="%s"}`
+	tidbSumCPUQuotaMetricsPattern = `tidb_server_maxprocs{cluster="%s"}`
+	queryPath                     = "/api/v1/query"
+	statusSuccess                 = "success"
+
+	httpRequestTimeout = 5
```

Comment:
I prefer to use `httpRequestTimeout = 5 * time.Second` here. Once we don't use second as the unit, we can just change it.